### PR TITLE
introduce embedded kafka server integration tests

### PR DESCRIPTION
**Pull request by talevy**

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,28 +1,41 @@
...
+  Dir.chdir runtime_jar_target
+  puts "Will download #{kafka_clients_url}"
+  `curl #{kafka_clients_url} -o kafka-clients-#{kafka_version}.jar`
   puts "Will download #{slf4j_url}"
   `curl #{slf4j_url} -o slf4j-api-#{slf4j_version}.jar`
   puts "Will download #{slf4j_nop_url}"
   `curl #{slf4j_nop_url} -o slf4j-noop-#{slf4j_version}.jar`
+  `echo #{home_dir}`
+  Dir.chdir home_dir
+  `lock_jars --vendor-dir #{test_jar_target}`
```
Comment (talevy) [inline Rakefile]: if you want to use the executable lock_jars then install ruby-maven gem before using lock_jars

**Comment by joekiller** [comment]
Hey guys y'all are doing a ton of work to get those jar dependencies when jar-dependencies should do it all for you. I know there was a bug (https://github.com/mkristian/jar-dependencies/issues/38) but it should since be fixed where you make jar-dependencies a development time requirement, it'll generate the jar include and then you download it.

I know you are moving away from jruby-kafka but I got CI tests working on it recently which may help you with the dependency part.

With jar-dependencies 0.3.2 and up you can do stuff like

```
  s.requirements << "jar 'org.apache.kafka:kafka_2.11', '0.9.0.0'"
  s.requirements << "jar 'org.slf4j:slf4j-log4j12', '1.7.13'"

  s.add_development_dependency 'jar-dependencies', "~> 0.3.2"
```

and then in rake

```
require 'jars/installer'
task :install_jars do
  Jars::Installer.vendor_jars!
end
```

and that's it!

**Comment by talevy** [comment]
thanks @joekiller!

I have checked that out, the reason I have not migrated to it just yet is because
we are have not finalized how we wish to take care of the jar dependencies. We have adopted the manual route for `logstash-output-elasticsearch`, and I was testing using `jar-dependencies` for this plugin's tests. and I couldn't figure out a way to separate the two. I am leaning towards re-introducing jar-dependencies as the default mechanism.

do you know how to separate test jars and runtime jars using jar-dependencies?

**Comment by joekiller** [comment]
It looks like scope settings are supported https://github.com/mkristian/jar-dependencies/search?p=3&q=scope&utf8=%E2%9C%93 but I'm unsure how to trigger it.

**Comment by talevy** [comment]
@joekiller thanks, I'll try it out!

**Comment by talevy** [comment]
Closing because progress has gone stale

**Pull request closed without merging**